File: android/feature/list/src/main/java/com/superlist/list/RetryBannerViewModel.kt
package com.superlist.list

data class RetryBannerState(
    val visible: Boolean = false,
    val message: String = "",
)

class RetryBannerViewModel {
    var state: RetryBannerState = RetryBannerState()
        private set

    fun showPersistentFailure() {
        state = RetryBannerState(true, "שגיאת סנכרון מתמשכת, ננסה שוב")
    }

    fun clear() {
        state = RetryBannerState()
    }
}

File: ios/Features/SharedList/RetryBannerViewModel.swift
import Foundation

@MainActor
final class RetryBannerViewModel: ObservableObject {
    @Published var isVisible: Bool = false
    @Published var message: String = ""

    func showPersistentFailure() {
        isVisible = true
        message = "שגיאת סנכרון מתמשכת, ננסה שוב"
    }

    func clear() {
        isVisible = false
        message = ""
    }
}
